File: sdks/javascript/src/transform.ts
/**
 * Transform component for position, rotation, and scale.
 */

import { Vec3 } from './math';

/**
 * Transform component for position, rotation, and scale.
 */
export class Transform {
  /** Position of the transform */
  position: Vec3;
  
  /** Rotation of the transform */
  rotation: Vec3;
  
  /** Scale of the transform */
  scale: Vec3;

  /**
   * Create a new transform.
   * 
   * @param position - Initial position
   * @param rotation - Initial rotation
   * @param scale - Initial scale
   */
  constructor(
    position: Vec3 = Vec3.zero(),
    rotation: Vec3 = Vec3.zero(),
    scale: Vec3 = Vec3.one()
  ) {
    this.position = position;
    this.rotation = rotation;
    this.scale = scale;
  }

  toString(): string {
    return `Transform(pos=${this.position}, rot=${this.rotation}, scale=${this.scale})`;
  }
}

File: sdks/python/windjammer_sdk/transform.py
"""Transform component for position, rotation, and scale."""

from .math import Vec3, Quat


class Transform:
    """Transform component for position, rotation, and scale."""
    
    def __init__(
        self,
        position: Vec3 = None,
        rotation: Vec3 = None,
        scale: Vec3 = None
    ):
        self.position = position or Vec3.zero()
        self.rotation = rotation or Vec3.zero()
        self.scale = scale or Vec3.one()
    
    def __repr__(self) -> str:
        return f"Transform(pos={self.position}, rot={self.rotation}, scale={self.scale})"

